src/main/java/org/broadinstitute/hellbender/engine/dataflow/transforms/PairReadsAndVariants.java

```diff
@@ -0,0 +1,87 @@
+package org.broadinstitute.hellbender.engine.dataflow.transforms;
+
+import com.google.cloud.dataflow.sdk.transforms.*;
+import com.google.cloud.dataflow.sdk.transforms.join.CoGbkResult;
+import com.google.cloud.dataflow.sdk.transforms.join.CoGroupByKey;
+import com.google.cloud.dataflow.sdk.transforms.join.KeyedPCollectionTuple;
+import com.google.cloud.dataflow.sdk.values.KV;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.cloud.dataflow.sdk.values.TupleTag;
+import org.broadinstitute.hellbender.engine.dataflow.datasources.VariantShard;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.read.GATKRead;
+import org.broadinstitute.hellbender.utils.variant.Variant;
+
+/**
+ * PairReadsAndVariants takes two PCollections (GATKRead and Variant) and create a single PCollection of KV where
+ * GATKRead is the key and Variant is the value. We do this join by sharding both collections by "variant shard" and
+ * checking for overlap on each "shard."
+ * This currently will not work with unmapped reads.
```

Comment:
This could be made easier to read by first finishing to say **what** it does before starting on **how**. 

Suggested rephrasing:

"PairReadsAndVariants takes two PCollections (GATKRead and Variant) and returns a  PCollection with a (GATKRead,Variant) pair for every read and variant that overlap. This means that a read or variant may be present multiple times in the output.

The reads currently must all be mapped.

The function works by (...)"